manifests/plugin/discovery.pp

```diff
@@ -4,44 +4,50 @@
 #
 # === Parameters:
 #
-# $version::         version string of discovery image, in form of x.y.z-r
+# $major_version::   version string of discovery image, in form of x.y
 #
-# $source::          mirror url from which the image files should be obtained, you
-#                    can use http(s):// or file://
+# $minor_version::   version string of discovery image, in form of x.y.z
 #
-# $initrd::          name of initrd image file
+# $source::          mirror url from which the image files should be obtained, you
+#                    can use http(s):// or file:// (include the trailing slash)
 #
-# $kernel::          name of kernel file
+# $image_name::      name of the image tarball file
 #
 # $install_images::  should the installer download and setup discovery images
 #                    for you? the average size is few hundreds of MB
 #                    type:boolean
 #
 class foreman::plugin::discovery (
-  $version        = $foreman::plugin::discovery::params::version,
-  $source         = $foreman::plugin::discovery::params::source,
-  $initrd         = $foreman::plugin::discovery::params::initrd,
-  $kernel         = $foreman::plugin::discovery::params::kernel,
+  $major_version  = $foreman::plugin::discovery::params::major_version,
+  $minor_version  = $foreman::plugin::discovery::params::minor_version,
   $install_images = $foreman::plugin::discovery::params::install_images,
+  $tftp_root      = $foreman::plugin::discovery::params::tftp_root,
 ) inherits foreman::plugin::discovery::params {
 
   validate_bool($install_images)
 
+  $source          = "http://downloads.theforeman.org/discovery/releases/${major_version}/"
```

Comment:
Hmmm so what do you expect? Can you list me inputs you want to see? Because I have just withdrawn this from options. I don't understand what you mean. Thanks.